shell.nix

```diff
@@ -0,0 +1,18 @@
+{ withHoogle ? true
+}:
+let
+  default = import ./default.nix {};
+in
+default.nix-tools._raw.shellFor {
+  packages    = p: map (x: p."${x}") [
+    "io-sim"
+    "io-sim-classes"
+    "ouroboros-consensus"
+    "ouroboros-network"
+    "typed-transitions"
+  ];
+  withHoogle  = withHoogle;
+  buildInputs = with default.nix-tools._raw; [
+    cabal-install.components.exes.cabal
```

Comment:
I think it's fine to assume that the user already has the latest cabal in their profile, so we don't need to add it to the project's dependencies.

Reply:
Is it really? The cabal in question might be incongruent in some fashion with the GHC in use -- or even completely absent.